src/libsyntax/feature_gate.rs

```diff
@@ -1599,6 +1599,9 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
                     gate_feature_post!(&self, extern_types, i.span,
                                        "extern types are experimental");
             }
+            ast::ForeignItemKind::Macro(..) => {
+                // FIXME: do we allow macro invocs in `extern {}`?
```

Comment:
Yes, this should be feature gated.

Reply:
Yeah, the feature gate is checked in the expansion code since this visitor is post-expansion.

Reply:
Thought it was, at least. Must have forgotten to push some changes from my laptop.